lib/services/project-data-service.ts

```diff
@@ -119,8 +119,10 @@ export class ProjectDataService implements IProjectDataService {
 		this.projectDataCache[projectDir] =
 			this.projectDataCache[projectDir] ||
 			this.$injector.resolve<IProjectData>(ProjectData);
-		this.projectDataCache[projectDir].initializeProjectData(projectDir);
-
+		if (!this.projectDataCache[projectDir].initialized) {
+			this.projectDataCache[projectDir].initializeProjectData(projectDir);
+			this.projectDataCache[projectDir].initialized = true;
+		}
```

Comment:
I think we should move this check into `initializeProjectData()` in `lib/project-data.ts` - that way the callers don't need to do any checking - just call `initializeProjectData()`.